Import libarchive-3.6.1-4.fc38.src.rpm as part of XS-1557

glehmann opened the pull request:
* update for openssl3
scratch build: https://koji.xcp-ng.org/taskinfo?taskID=89633


Update  from @rzr:

- [x] Koji pre-build XCPNG2710 (v8.3-u-pcoval2): https://koji.xcp-ng.org/taskinfo?taskID=94080

stormi reviewed (changes requested):
We need to update branch `upstream` (and rename it to `upstream-8.3` in the process) with the upstream package XS based their upon. And then to decide whether we want to go with XenServer's packaging or follow the upstream one instead.

rzr reviewed (approved):
Approving for @glehmann 

May I rebase on upstream ?

rzr committed:
Import libarchive-3.6.1-4.fc38.src.rpm

Signed-off-by: Philippe Coval <philippe.coval@vates.tech>
Origin: https://archives.fedoraproject.org/pub/archive/fedora/linux/releases/38/Everything/source/tree/Packages/l/libarchive-3.6.1-4.fc38.src.rpm

rzr committed:
Rebuild with openssl-3

It looks like libarchive-3.6.1-2.xs8.src.rpm
does not bring any value for xcp-ng compared to upstream package:
libarchive-3.6.1-4.fc38.src.rpm

Only XS metadata is added (hash, version), and history is loss.
Better align to upstream to ease maintainability.

Origin: https://github.com/xcp-ng-rpms/libarchive/pull/1
Signed-off-by: Philippe Coval <philippe.coval@vates.tech>

rzr committed:
Merge branch 'upstream-8.3' into 8.3-XS-1557

It looks like libarchive-3.6.1-2.xs8.src.rpm
does not bring any value for xcp-ng compared to upstream package:
libarchive-3.6.1-4.fc38.src.rpm

Only XS metadata is added (hash, version), and history is loss.
Better align to upstream to ease maintainability.

Conflicts lines are reintroduced for next version.

RPM forbids non temporal order, so entries are nested.

Conflicts: SPECS/libarchive.spec
Origin: https://github.com/xcp-ng-rpms/libarchive/pull/1
Signed-off-by: Philippe Coval <philippe.coval@vates.tech>

stormi reviewed (approved):
Comparison with XS-8.3 shows differences that are easy to interpret, and naturally caused by the fact that we chose to rebase on the original SRPM that they forked.

We gain in the process a call to ldconfig upon installation and uninstallation, that the XS package is apparently missing.

rzr commented:
- [x] Koji build (v8.3-incoming): https://koji.xcp-ng.org/taskinfo?taskID=97799
